main.go

```diff
@@ -161,65 +167,118 @@ func NewNSMClient(ctx context.Context, rootConf *config.Config) networkservice.N
 		spiffejwt.TokenGeneratorFunc(source, rootConf.MaxTokenLifetime),
 		clientCC,
 		sendfd.NewClient(),
+		token.NewClient(),
 	)
 }
 
+// Connection is a return type for RunClient
+type Connection struct {
+	Client     networkservice.NetworkServiceClient
+	Connection *networkservice.Connection
+	Cancel     context.CancelFunc
+}
+
 // RunClient - runs a client application with passed configuration over a client to Network Service Manager
-func RunClient(ctx context.Context, rootConf *config.Config, nsmClient networkservice.NetworkServiceClient) ([]*networkservice.Connection, error) {
+func RunClient(ctx context.Context, rootConf *config.Config, nsmClient networkservice.NetworkServiceClient) ([]*Connection, error) {
 	// Validate config parameters
 	if err := rootConf.IsValid(); err != nil {
-		return []*networkservice.Connection{}, err
+		return nil, err
+	}
+
+	var requestConfigs []*requestConfig
+	for i := range rootConf.NetworkServices {
+		nsConf := &rootConf.NetworkServices[i]
+		if err := nsConf.MergeWithConfigOptions(rootConf); err != nil {
+			log.Entry(ctx).Errorf("error during nsmClient config aggregation: %v", err)
+			return nil, err
+		}
+		requestConfigs = appendNSConf(nsConf, requestConfigs)
 	}
 
 	// ********************************************************************************
 	// Initiate connections
 	// ********************************************************************************
 
 	// A list of cleanup operations
-	connections := []*networkservice.Connection{}
-
-	for idx, clientConf := range rootConf.NetworkServices {
-		err := clientConf.MergeWithConfigOptions(rootConf)
-		if err != nil {
-			log.Entry(ctx).Errorf("error during nsmClient config aggregation %v", err)
-			return connections, err
-		}
-		// We need update
-		outgoingMechanism := &networkservice.Mechanism{
-			Cls:        cls.LOCAL,
-			Type:       clientConf.Mechanism,
-			Parameters: map[string]string{},
-		}
-		switch clientConf.Mechanism {
-		case kernel.MECHANISM:
-			outgoingMechanism.Parameters[common.InterfaceNameKey] = clientConf.Path[0]
-			fileURL := &url.URL{Scheme: "file", Path: "/proc/self/ns/net"}
-			kernel.ToMechanism(outgoingMechanism).SetNetNSURL(fileURL.String())
-
-		case memif.MECHANISM:
-			outgoingMechanism.Parameters[memif.SocketFilename] = path.Join(clientConf.Path...)
+	var connections []*Connection
+	for _, requestConfig := range requestConfigs {
+		var clients []networkservice.NetworkServiceClient
+		for _, nsConf := range requestConfig.nsConfs {
+			switch nsConf.Mechanism {
+			case kernelmech.MECHANISM:
+				clients = append(clients, kernel.NewClient(kernel.WithInterfaceName(nsConf.Path[0])))
+			case vfiomech.MECHANISM:
+				cgroupDir, err := cgroupDirPath()
+				if err != nil {
+					log.Entry(ctx).Errorf("failed to get devices cgroup: %v", err)
+					return connections, err
```

Comment:
As I can see in case of an error in this branch client will not fail. So then we should call `continue` here OR we should change the logic, that if one of the connections is failed then the client should be failed.

Reply:
OK, I guess we can start as many connections as possible and fail `RunClient` only if we fail all the connections. Replaced `return err` with `continue`.